include/evmc/evmc.h

```diff
@@ -34,20 +34,23 @@ enum
     EVMC_ABI_VERSION = 6
 };
 
+
 /**
- * Big-endian 256-bit integer.
+ * The fixed size array of 32 bytes.
  *
- * 32 bytes of data representing big-endian 256-bit integer. I.e. bytes[0] is
- * the most significant byte, bytes[31] is the least significant byte.
- * This type is used to transfer to/from the VM values interpreted by the user
- * as both 256-bit integers and 256-bit hashes.
+ * 32 bytes of data capable of storing e.g. 256-bit hashes.
  */
-struct evmc_uint256be
+struct evmc_bytes32
 {
-    /** The 32 bytes of the big-endian integer or hash. */
+    /** The 32 bytes. */
     uint8_t bytes[32];
 };
 
+/**
+ * The alias for evmc_bytes32 to represent a big-endian 256-bit integer.
+ */
+typedef struct evmc_bytes32 evmc_uint256be;
```

Comment:
So I am thinking perhaps we should have `struct __evmc_bytes32 {..}` and then two typedefs (and the same applies to `evmc_address`) so that it is consistent requiring (or not) `struct` in further use cases.

Right now it is weird that one has to use `struct evmc_bytes32 a;` and `evmc_uint256be a;`

Reply:
Also wonder if we do have the shared underlying type, operators applied to them propagate to the two typedefs?

Reply:
These are the same types effectively. Yes, it feels weird to me too.